core/arxiv/submission/schedule.py

```diff
@@ -0,0 +1,66 @@
+"""
+Policies for announcement scheduling.
+
+Submissions to arXiv are normally made public on Sunday through Thursday, with
+no announcements Friday or Saturday.
+
++-----------------------+----------------+------------------------------------+
+| Received Between (ET) | Announced (ET) | Mailed                             |
++=======================+================+====================================+
+| Mon 14:00 - Tue 14:00 | Tue 20:00      | Tuesday Night / Wednesday Morning  |
+| Tue 14:00 - Wed 14:00 | Wed 20:00      | Wednesday Night / Thursday Morning |
+| Wed 14:00 - Thu 14:00 | Thu 20:00      | Thursday Night / Friday Morning    |
+| Thu 14:00 - Fri 14:00 | Sun 20:00      | Sunday Night / Monday Morning      |
+| Fri 14:00 - Mon 14:00 | Mon 20:00      | Monday Night / Tuesday Morning     |
++-----------------------+----------------+------------------------------------+
+
+"""
+
+from typing import Optional
+from datetime import datetime, timedelta
+from enum import IntEnum
+from pytz import timezone, UTC
+
+ET = timezone('US/Eastern')
+
+Weekdays = IntEnum('Weekdays', 'Mon Tue Wed Thu Fri Sat Sun', start=1)
+
+ANNOUNCE_TIME = 20
+FREEZE_TIME = 13
+
+WINDOWS = [
+    ((Weekdays.Fri - 7, 14), (Weekdays.Mon, 14), (Weekdays.Mon, 20)),
+    ((Weekdays.Mon, 14), (Weekdays.Tue, 14), (Weekdays.Tue, 20)),
+    ((Weekdays.Tue, 14), (Weekdays.Wed, 14), (Weekdays.Wed, 20)),
+    ((Weekdays.Wed, 14), (Weekdays.Thu, 14), (Weekdays.Thu, 20)),
+    ((Weekdays.Thu, 14), (Weekdays.Fri, 14), (Weekdays.Sun, 20)),
+    ((Weekdays.Fri, 14), (Weekdays.Mon + 7, 14), (Weekdays.Mon + 7, 20)),
+]
+
+
+def _datetime(ref: datetime, isoweekday: int, hour: int) -> datetime:
+    days_hence = isoweekday - ref.isoweekday()
+    repl = dict(hour=hour, minute=0, second=0, microsecond=0)
+    return (ref + timedelta(days=days_hence)).replace(**repl)
+
+
+def next_announcement_time(ref: Optional[datetime] = None) -> datetime:
```

Comment:
We'll want to make this holiday-aware at some point. Probably OK for now.

Reply:
https://arxiv-org.atlassian.net/browse/ARXIVNG-2191

Reply:
@mhl10 + @erickpeirson `pandas` has a US federal holidays if you're willing to have that as a dependency:

```
(scipy-dev) Lucass-MacBook:scipy rlucas$ python
Python 3.6.7 |Anaconda, Inc.| (default, Oct 23 2018, 14:01:38) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from pandas.tseries.holiday import USFederalHolidayCalendar
>>> cal = USFederalHolidayCalendar()
>>> holidays = cal.holidays(start='2020-01-01', end='2020-12-31').to_pydatetime()
>>> for i in holidays:
...     print(i)
... 
2020-01-01 00:00:00
2020-01-20 00:00:00
2020-02-17 00:00:00
2020-05-25 00:00:00
2020-07-03 00:00:00
2020-09-07 00:00:00
2020-10-12 00:00:00
2020-11-11 00:00:00
2020-11-26 00:00:00
2020-12-25 00:00:00
```

or did this mean something customizeable that would exist as part of the microservice and could be extended for ArXiv specific holiday needs (e.g. other countries or holidays too)?

Reply:
The 'holidays' are days that arXiv would normally announce new papers but does not due to the staff being on holiday. It should probably be named 'announcement_skipped_days'. 

> or did this mean something customizeable that would exist as part of the microservice and could be extended for ArXiv specific holiday needs (e.g. other countries or holidays too)?